src/circ_deque.jl

```diff
@@ -80,6 +80,11 @@ end
     ret
 end
 
+# Iteration via getindex
+Base.start(d::CircularDeque) = 1
+Base.next(d::CircularDeque, i) = (d[i], i+1)
+Base.done(d::CircularDeque, i) = i == length(d) + 1
```

Comment:
Using `getindex` is a good first pass.  It does do a number of checks that might (?) not be necessary during iteration, so if you wanted, you could try to implement a more complicated version of these functions which access the buffer directly. Not sure if the actual benefit.